nodejs/awsx/apigateway/api.ts

```diff
@@ -581,6 +624,74 @@ function addEventHandlerRouteToSwaggerSpec(
     }
 }
 
+function addAuthorizersToSwagger(swagger: SwaggerSpec, authorizers: authorizer.LambdaAuthorizerDefinition[]): string[] {
+    const authNames: string[] = [];
+    swagger["securityDefinitions"] = swagger["securityDefinitions"] || {};
+
+    for (const auth of authorizers) {
+        const suffix = Object.keys(swagger["securityDefinitions"]).length;
+        const authName = auth.authorizerName || `authorizer-${suffix}`;
+
+        const securityDef: SecurityDefinition = {
+            type: "apiKey",
+            name: auth.parameterName,
+            in: auth.parameterLocation,
+            "x-amazon-apigateway-authtype": auth.authType,
+            "x-amazon-apigateway-authorizer": getLambdaAuthorizer(authName, auth),
+        };
+
+        // Add security definition if it's a new authorizer
+        if (!swagger["securityDefinitions"][authName]) {
+            authNames.push(authName);
+            swagger["securityDefinitions"][authName] = securityDef;
+        } else if (swagger["securityDefinitions"][authName] !== securityDef) {
```

Comment:
this doesn't seem necessary.  either we have a collision on authName or not.  If we don't, we're good.  Otehrwise, we always throw.  

In other words, since you just defined securityDef above, if `!swagger["securityDefinitions"][authName]` is false, then `swagger["securityDefinitions"][authName] !== securityDef` will always be true, since hte existing sec-def will never equal `securityDef`